sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -187,9 +189,14 @@ class HashJoinIterator(
     val buildSide: GpuBuildSide,
     val compareNullsEqual: Boolean, // This is a workaround to how cudf support joins for structs
     private val spillCallback: SpillCallback,
+    private val condition: Option[Expression],
+    private val fallback: Option[
+        (Iterator[ColumnarBatch], Iterator[ColumnarBatch]) => Iterator[ColumnarBatch]],
+    private val joinOutputRows: GpuMetric,
```

Comment:
Not for this PR, but wondering if we should have a case class or something to help organize the metrics and/or join key/attr params to try to keep the number of parameters to this and related join code reasonable.